cohort-evaluator/src/test/java/com/ibm/cohort/cql/version/ResourceSelectorTest.java

```diff
@@ -0,0 +1,145 @@
+                createObject("1", "1.0.1"),
+                createObject("2", "1.0.2"),
+                expected,
+                createObject("4", "1.0.4"),
+                createObject("5", "1.0.5")
+        );
+
+        ResourceSelector<SimpleObject> selector = getSelector();
+        SimpleObject actual = selector.selectSpecificVersionOrLatest(objects, version);
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void selectSpecificVersionOrLatest_noVersion() {
+        String id = "5";
+        String version = "1.0.5";
+        SimpleObject expected = createObject(id, version);
+
+        List<SimpleObject> objects = Arrays.asList(
+                createObject("0", "1.0.0"),
+                createObject("1", "1.0.1"),
+                createObject("2", "1.0.2"),
+                createObject("3", "1.0.3"),
+                createObject("4", "1.0.4"),
+                expected
+        );
+
+        ResourceSelector<SimpleObject> selector = getSelector();
+        SimpleObject actual = selector.selectSpecificVersionOrLatest(objects, null);
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void selectSpecificVersionOrLatest_versionNotFound() {
+        List<SimpleObject> objects = Arrays.asList(
+                createObject("0", "1.0.0"),
+                createObject("1", "1.0.1"),
+                createObject("2", "1.0.2"),
+                createObject("3", "1.0.3"),
+                createObject("4", "1.0.4"),
+                createObject("5", "1.0.5")
+        );
+
+        ResourceSelector<SimpleObject> selector = getSelector();
+        SimpleObject actual = selector.selectSpecificVersionOrLatest(objects, "9.9.99");
+        Assert.assertNull(actual);
+    }
+
+    @Test
+    public void selectSpecificVersionOrLatest_noObjects() {
+        List<SimpleObject> objects = Collections.emptyList();
+        ResourceSelector<SimpleObject> selector = getSelector();
+        SimpleObject actual = selector.selectSpecificVersionOrLatest(objects, "1.0.0");
+        Assert.assertNull(actual);
+    }
+
+    @Test
+    public void selectSpecificVersionOrLatest_specificVersion_duplicateVersions() {
+        String id = "3";
+        String version = "1.0.3";
+        SimpleObject expected = createObject(id, version);
+
+        List<SimpleObject> objects = Arrays.asList(
+                createObject("0", "1.0.0"),
+                createObject("1", version),
+                createObject("2", "1.0.2"),
+                expected,
+                createObject("4", "1.0.4"),
+                createObject("5", "1.0.5")
+        );
+
+        ResourceSelector<SimpleObject> selector = getSelector();
+        Assert.assertThrows(
+                IllegalArgumentException.class,
+                () -> selector.selectSpecificVersionOrLatest(objects, version)
+        );
+    }
+
+    /*
+     * I'm not sure if this is desirable behavior since duplicates cause an
```

Comment:
Agree with the behavior seeming off.

Reply:
Per our standup discussions, I'll make a ticket on this behavior.